test/robustness/overlay/strategies/intersection_points_and_sides.cpp

```diff
@@ -0,0 +1,332 @@
+    auto const dq2 = bg::distance(ip, q2);
+
+    // Calculate the azimuth (0 = North) and modify it to Mathematical (0 = east)
+    auto const azimuth_p = bg::azimuth(ip, dp1 < dp2 ? p1 : p2);// - bg::math::pi<T>() / 2.0;
+    auto const azimuth_q = bg::azimuth(ip, dq1 < dq2 ? q1 : q2);// - bg::math::pi<T>() / 2.0;
+
+    // Calculate the average circular azimuth
+    auto const sum_sin = std::sin(azimuth_p) + std::sin(azimuth_q);
+    auto const sum_cos = std::cos(azimuth_p) + std::cos(azimuth_q);
+    auto const azimuth_avg = std::atan2(sum_sin, sum_cos);
+
+    auto const d = std::numeric_limits<T>::epsilon() * epsilon_multiplier;
+
+    // Calculate the point using sin,cos in this order (because North=0)
+    Point result;
+    bg::set<0>(result, bg::get<0>(ip) + d * std::sin(azimuth_avg));
+    bg::set<1>(result, bg::get<1>(ip) + d * std::cos(azimuth_avg));
+
+    // T const rd = 180 / bg::math::pi<T>();
+    // std::cout << "Index " << azimuth_p * rd << " " << azimuth_q * rd << " -> " << azimuth_avg * rd << " dist=" << bg::distance(e, ip) << " d=" << d << std::endl;
+
+    return result;
+}
+
+template <typename T, typename Settings>
+void test_all(Settings const& settings)
+{
+    using point_t = bg::model::d2::point_xy<T>;
+    using segment_type = bg::model::referring_segment<const point_t>;
+    using result_type = bg::segment_intersection_points<point_t> ;
+
+    using points_policy_type = bg::policies::relate::segments_intersection_points
+        <
+            result_type
+        >;
+
+    using side_by_triangle = bg::strategy::side::side_by_triangle<T>;
+    using side_non_robust = bg::strategy::side::side_non_robust<T>;
+    using side_robust_fp_3 = bg::strategy::side::side_robust<T, bg::strategy::side::fp_equals_policy, 3>;
+    using side_robust_eq_3 = bg::strategy::side::side_robust<T, bg::strategy::side::epsilon_equals_policy, 3>;
+    using side_rounded_input = bg::strategy::side::side_rounded_input<T>;
+
+    auto const t0 = std::chrono::high_resolution_clock::now();
+
+    std::size_t count = 0;
+    std::size_t errors_triangle = 0;
+    std::size_t errors_non_robust = 0;
+    std::size_t errors_robust_fp_3 = 0;
+    std::size_t errors_robust_eq_3 = 0;
+    std::size_t errors_rounded = 0;
+    std::size_t errors_bsp_triangle = 0;
+    std::size_t errors_bsp_non_robust = 0;
+    std::size_t errors_bsp_robust_fp_3 = 0;
+    std::size_t errors_bsp_robust_eq_3 = 0;
+    std::size_t errors_bsp_rounded = 0;
+
+    std::mt19937 gen(settings.seed);
+    std::uniform_real_distribution<T> dis(0, settings.size);
+
+    T sum_distance_error = 0;
+    T sum_distance_for_rounded_error = 0;
+
+    for (std::size_t index = 0; index < settings.count; index++)
+    {
+        const point_t p1 = random_point<point_t>(gen, dis);
+        const point_t p2 = random_point<point_t>(gen, dis);
+        const point_t q1 = random_point<point_t>(gen, dis);
+        const point_t q2 = random_point<point_t>(gen, dis);
```

Comment:
Random points probably shouldn't be usen in tests because various libraries may implement random generator differently so we may be potentially testing agsinst different points.